### ISSUE TITLE
useDefaultRoute is not working from v6.7.0

### ISSUE THREAD

**Issue by almandsky** [brightcove/hot-shots#136]
https://github.com/brightcove/hot-shots/compare/v6.6.0...v6.7.0

Seems like the `createTransport` is taking the `host` parameter before it is updated by the `useDefaultRoute`.  That means, from v6.7.0, if the hot-shots is running inside a docker container, it will still send the metrics to localhost by default, instead of sending to the actual host ip `172.17.0.1`.

We shall consider move the code of line 146~152 to before line 43, which is where the `host` is being used for `createTransport`.

https://github.com/brightcove/hot-shots/blob/master/lib/statsd.js#L146-L152

**Comment by Gobie** [comment]
`cacheDns` option doesn't work neither because of the same problem.

**Comment by almandsky** [comment]
Before the v6.7.0, the `Client.prototype.sendMessage` will use the `host` args after the init is completed.

But from v6.7.0, the `host` is passed to the `createTransport` during the init, so the `host` override will not work after the `this.socket` is initialized via `createTransport`.

https://github.com/brightcove/hot-shots/compare/v6.5.1...v6.8.1#diff-cfdf1e0afcbe7fb32697031bec3cc7deL393

**Comment by almandsky** [comment]
@Gobie 

I include the fix for `cacheDns` in my PR as well.

**Comment by Gobie** [comment]
Hey @almandsky, thanks for taking the time to tackle this issue.
But I believe `cacheDns` still remains broken as `dns.lookup` is an async call and by the time it is resolved `createTransport` have already been called and `host` can't be overridden.

**Comment by almandsky** [comment]
@Gobie 

I don't think `dns.lookup` is a async call.  It is still a sync call, but you can provide the callback function to it.

https://git.softwareadvice.com/qa/testcafe/blob/db29579085112bfa27808bbe680e2d3747c42ad8/node_modules/@types/node/dns.d.ts#L30

**Comment by Gobie** [comment]
I would respectfully disagree
```js
const dns = require('dns')
dns.lookup('google.com', console.log.bind(console, 'lookup callback'))
console.log('call after lookup')
```

```bash
➜ node jest/test.js
call after lookup
lookup callback null 216.58.201.110 4
```

**Comment by almandsky** [comment]
@Gobie 

Sorry that I am wrong.  I read some posts before saying that the dns.lookup was blocking the main process.  Thanks for the correction.

In this case, yes, the `cacheDns` fix may required to update the socket host or re-create the new socket with the updated host in the callback of the `dns.lookup`.

**Comment by bdeitte** [comment]
Creating a new ticket on cacheDns, good to know.

**Comment by bdeitte** [comment]
This one fixed in 6.8.2
